File: src/main/java/com/ouertatani/gestionproduit/model/CommandeProduit.java
package com.ouertatani.gestionproduit.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.*;
import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CommandeProduit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "commande_id", nullable = false)
    @JsonBackReference  // ← empêche la boucle avec Commande
    private Commande commande;

    @ManyToOne
    @JoinColumn(name = "produit_id", nullable = false)
    private Produit produit;

    private int quantite;

    private BigDecimal prixUnitaire;
}

File: lib/models/commande_produit.dart
import 'produit.dart';

class CommandeProduit {
  final int? id;
  final int? commandeId;
  final int produitId;
  final int quantite;
  final double prixUnitaire;
  final Produit? produit;

  CommandeProduit({
    this.id,
    this.commandeId,
    required this.produitId,
    required this.quantite,
    required this.prixUnitaire,
    this.produit,
  });

  factory CommandeProduit.fromJson(Map<String, dynamic> json) {
    final produitJson = json['produit'];
    return CommandeProduit(
      id: json['id'],
      commandeId: json['commande'] != null ? json['commande']['id'] : null,
      produitId: produitJson != null ? produitJson['id'] : json['produitId'],
      quantite: json['quantite'],
      prixUnitaire: (json['prixUnitaire'] as num).toDouble(),
      produit: produitJson != null ? Produit.fromJson(produitJson) : null,
    );
  }
}
